christfellowshipchurch/apollos-app
# [Chat] Chat Cleanup - Round 2

### Summary

This PR fixes the following random issues in Chat.

1. Layout broken when opening livestream in landscape ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061405134])).

1. Chat channel doesn't change when switching livestreams ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061357301)).

1. Hide DMs from livestream to livestream (filter by last message date) ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3065613273)).

1. Fixes this navigation header issue on DM Channel view (see pic).

<img width="400" src="https://user-images.githubusercontent.com/200488/94868853-f9fd1500-0411-11eb-8193-b8269533026c.gif">


### Discussion

This PR has a few other commits.

1. There is a DEV version and a PROD version of the Zoom SDK that gets linked up at build time. The PROD version got committed to git, so this PR reverts that back to the DEV version for version control.

1. The Stream.io component was using a lite version of `moment` called `dayjs`. Let's just make this use `moment` because the app already uses `moment` and we can avoid bundling `dayjs` if we don't import it.

1. There is a commit related to [[Bug] Background audio doesn't work when locking phone](https://3.basecamp.com/3926363/buckets/17566548/todos/3061358681), but I reverted it. You can read the comments in that todo but the issue is not valid. But I want to keep this commit in the history just in case.

### Testing

(Same items as above.)

1. Turn the device to landscape. Open a livestream. Video should be fullscreen and Chat should be hidden. (Returning to portrait should display chat.)

1. With two livestreams going, open one and chat. Close the livestream. Open the other one. Chat should be different.

1. This one is a little harder. You need to have some DMs with a user that is older than _X amount of time_ and some with a user that are not. Those that are within _X amount of time_ will show up, first in the blue Messages Banner and then in the DM Channels List. Those that are not will not show up anywhere. From the user with older messages, send a new message. The channel will show up in the banner and in the Channels List. You can hard code the _X amount of time_ in two places. See the commit for where.

1. Navigate to a 1-on-1 DM channel. Ensure the header and footer look correct.
